### PR TITLE
add NEVER_UNIT automation

### PR DESCRIPTION
**Pull request by smilerz**

### CODE REVIEW

File: cookbook/helper/ingredient_parser.py [hunk 1]
```diff
@@ -50,10 +51,22 @@ def __init__(self, request, cache_mode, ignore_automations=False):
...
                     self.never_unit[a.param_1] = a.param_2
                 caches['default'].set(NEVER_UNIT_CACHE_KEY, self.never_unit, 30)
+
+            TRANSPOSE_WORDS_CACHE_KEY = f'automation_transpose_words_{self.request.space.pk}'
+            if c := caches['default'].get(TRANSPOSE_WORDS_CACHE_KEY, None):
+                self.transpose_words = c
+                caches['default'].touch(TRANSPOSE_WORDS_CACHE_KEY, 30)
+            else:
+                i = 0
+                for a in Automation.objects.filter(space=self.request.space, disabled=False, type=Automation.TRANSPOSE_WORDS).only('param_1', 'param_2').order_by('order').all():
```
Comment (vabene1111) [inline cookbook/helper/ingredient_parser.py:61]: is transpose_words and never_unit mixed up here ?

File: cookbook/helper/ingredient_parser.py [hunk 2]
```diff
@@ -249,15 +262,31 @@ def apply_never_unit_automations(self, tokens):
...
-            return tokens
+            return ingredient
+
+        else:
+            if self.transpose_words:
+                for rule in self.transpose_words:
+                    ingredient = re.sub(rf"\b({rule[0]}) ({rule[1]})\b", r"\2 \1", ingredient)
+
+            else:
+                for rule in Automation.objects.filter(space=self.request.space, type=Automation.TRANSPOSE_WORDS, disabled=False).order_by('order'):
```
Comment (vabene1111) [inline cookbook/helper/ingredient_parser.py:286]: this hits the database for every ingredient parsed and might also execute extremly large amounts of regex

File: docs/features/automation.md [hunk 3]
```diff
@@ -68,6 +68,17 @@ You can also create them manually by setting the following
 These rules are processed whenever you are importing recipes from websites or other apps
 and when using the simple ingredient input (shopping, recipe editor, ...).
 
+## Transpose Words
```
Comment (vabene1111) [inline docs/features/automation.md:71]: Is there also documentation for never unit ? or is that the same as transpose words?

File: cookbook/helper/ingredient_parser.py (hunk 2 above)
Comment (smilerz) [reply cookbook/helper/ingredient_parser.py:286]: That would only occur if not cached - I can take that out and have it so the rule never runs if the cache doesn't exist.  Or do you thoughts on another way to limit it?

File: docs/features/automation.md (hunk 3 above)
Comment (smilerz) [reply docs/features/automation.md:71]: they are both there...

File: cookbook/helper/ingredient_parser.py (hunk 1 above)
Comment (vabene1111) [reply cookbook/helper/ingredient_parser.py:63]: iteresting, when i first reviewed there was somethign else, now it look correct

**Comment by vabene1111** [comment]
ok so finally got this merged, thanks for your patience. I think one of the next things we need to do with automations is write some more/proper tests because i am slowly loosing oversight but other than that everything seems to be correct.

**Pull request merged**